### PR TITLE
Remove the deprecated option ``use_legacy_supplementaries``

### PR DESCRIPTION
**Pull request by bouweandela**
## Description

Remove the deprecated option use_legacy_supplementaries, scheduled for removal in v2.10.0 (see https://github.com/ESMValGroup/ESMValCore/pull/1609#Deprecations).

**Upgrade advice**

The recommended upgrade procedure is to remove `use_legacy_supplementaries` from config-user.yml (if it was there) and remove any mention of `fx_variables` from the recipe. If automatically defining the required supplementary variables does not work, define them in the variable or (additional_)datasets section as described in the [documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/recipe/overview.html#defining-supplementary-variables-ancillary-variables-and-cell-measures).

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [ ] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [ ] ~[🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)~
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

File: tests/integration/recipe/test_recipe.py [hunk 1]
```diff
@@ -2323,731 +2104,32 @@ def test_landmask_no_fx(tmp_path, patched_failing_datafinder, session):
...
-
         diagnostics:
           diagnostic_name:
             variables:
-              tas:
+              tos:
                 preprocessor: preproc
-                project: CMIP6
-                mip: Amon
+                project: CMIP7
```
Comment (valeriupredoi) [inline tests/integration/recipe/test_recipe.py:2118]: :rofl: this will come back and bite you very soon :grin:

**Review by valeriupredoi — APPROVED** [review]
is ESMValCore so light now that it can :airplane: ? I'd see about CMIP7 project, bud :grin:

File: tests/integration/recipe/test_recipe.py (hunk 1 above)
Comment (bouweandela) [reply tests/integration/recipe/test_recipe.py:2118]: Haha, yes, that's what I thought. It wasn't me who wrote that test.

**Comment by bouweandela** [comment]
Thanks for reviewing!

**Pull request merged**